Microsoft.Identity.Web/HttpContextExtensions.cs

```diff
@@ -0,0 +1,49 @@
+﻿using Microsoft.AspNetCore.Http;
+using System.IdentityModel.Tokens.Jwt;
+
+namespace Microsoft.Identity.Web
+{
+    public static class HttpContextExtensions
+    {
+        /// <summary>
+        /// Keep the validated token in associated with the Http request
+        /// </summary>
+        /// <param name="httpContext">Http context</param>
+        /// <param name="token">Token to preserve after the token is validated so that
+        /// it can be used in the actions</param>
+        public static void StoreTokenUsedToCallWebAPI(this HttpContext httpContext, JwtSecurityToken token)
+        {
+            httpContext.Items.Add("token", token);
+        }
+
+        /// <summary>
+        /// Get the parsed information about the token used to call the Web API
+        /// </summary>
+        /// <param name="httpContext">Http context associated with the current request</param>
+        /// <returns><see cref="JwtSecurityToken"/> used to call the Web API</returns>
+        public static JwtSecurityToken GetTokenUsedToCallWebAPI(this HttpContext httpContext)
+        {
+            return httpContext.Items["token"] as JwtSecurityToken;
```

Comment:
I have a question about this method. 

In the sample 2, we have a desktop app that calls a web api, that calls graph. There are two ATs involved, web api and graph. Currently, this method is setting only the web api's AT to the `httpContext.Items`. Is that the expected behavior?

Reply:
yes it is. The token used to call the Web API as we need its hash for the cache key in the case of OBO